src/System.Text.Json/src/System/Text/Json/Node/JsonNode.TraversalHelpers.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace System.Text.Json
+{
+    /// <summary>
+    ///   The base class that represents a single node within a mutable JSON document.
+    /// </summary>
+    public abstract partial class JsonNode
+    {
+        private static void AddToParent(
+            KeyValuePair<string, JsonNode> nodePair,
+            ref Stack<KeyValuePair<string, JsonNode>> currentNodes,
+            ref JsonNode toReturn,
+            DuplicatePropertyNameHandling duplicatePropertyNameHandling = DuplicatePropertyNameHandling.Replace)
+        {
+            if (currentNodes.Any())
```

Comment:
Can we just add some comment in here saying that currentNodes will really mean that we are inside either an object or an array?

Reply:
Ok, I can add it.